imgui/core.pyx

```diff
@@ -1935,6 +1935,220 @@ def menu_item(
     return clicked, inout_selected
 
 
+def open_popup(str name):
+    """Open a popup window.
+
+    Marks a popup window as open. Popups are closed when user click outside,
+    or activate a pressable item, or :func:`close_current_popup()` is
+    called within a :func:`begin_popup()`/:func:`end_popup()` block.
+    Popup identifiers are relative to the current ID-stack
+    (so :func:`open_popup` and :func:`begin_popup` needs to be at
+    the same level).
+
+    .. visual-example::
+        :title: Simple popup window
+        :height: 100
+        :width: 220
+        :auto_layout:
+
+        imgui.begin("Example: popup window")
+        show_popup = imgui.button("Open")
+        closable = True
+        if show_popup:
+            imgui.open_popup("Popup Title")
+        popup_opened, popup_visible = imgui.begin_popup_modal(
+            "Popup Title",
+            closable
+        )
+        if popup_opened:
+            imgui.text("Popup contents")
+            imgui.end_popup()
+        imgui.end()
+
+    Args:
+        name (str): label of the modal window.
+
+    .. wraps::
+        void OpenPopup(
+            const char* str_id
+        )
+    """
+    cimgui.OpenPopup(_bytes(name))
+
+
+def begin_popup(str name):
+    """Open a popup window.
+
+    Returns ``True`` if the popup is open and you can start outputting
+    content to it. Only call :func:`end_popup()` if :func:`begin_popup()`
+    returned true.
+
+    For practical example how to use this function, please see documentation
+    of :func:`open_popup`.
+
+    Args:
+        name (str): label of the modal window.
+
+    Returns:
+        opened (bool): True if popup is opened.
+
+    .. wraps::
+        bool BeginPopup(
+            const char* str_id
+        )
+    """
+    return cimgui.OpenPopup(_bytes(name))
+
+
+def begin_popup_modal(str title, closable=None, cimgui.ImGuiWindowFlags flags=0):
+    """Begin pouring popup contents.
+
+    For practical example how to use this function, please see documentation
+    of :func:`open_popup`.
+
+    Args:
+        title (str): label of the modal window.
+        closable (bool): define if popup is closable with X.
+        flags: Window flags. See:
+            :ref:`list of available flags <window-flag-options>`.
+
+    Returns:
+        tuple: ``(opened, visible)`` tuple of bools.
```

Comment:
I think we should reconsider the semantics and naming of `visible` output value and `closable` arg. I think it should be `visible` because allows to skip window display even if `OpenModal` was called. Note that there is a small bug later that allows to pass only `true` or `NULL` as the `p_open` argument.

I think that the example usage could to create a one-off modals like "message of the day" or warnings on first use of some application function.